omnigibson/utils/vision_utils.py

```diff
@@ -106,3 +106,65 @@ def segmentation_to_rgb(seg_im, N, colors=None):
         return (255.0 * use_colors[seg_im]).astype(np.uint8)
     else:
         return (use_colors[seg_im]).astype(np.float)
+
+class Remapper:
+    def __init__(self):
+        self.key_array = np.array([], dtype=np.uint32)  # Initialize the key_array as empty
+        self.known_ids = set()
+
+    def clear(self):
+        """Resets the key_array to empty."""
+        self.key_array = np.array([], dtype=np.uint32)
+        self.known_ids = set()
+
+    def remap(self, old_mapping, new_mapping, image):
+        """
+        Remaps values in the given image from old_mapping to new_mapping using an efficient key_array.
```

Comment:
Let's also add in a comment saying "if there are values in the image that are not found in old_mapping, we will map them to the "unlabeled""